### DISCUSSION TITLE
Cookie Session Refresh

### DISCUSSION THREAD

**Discussion by 1boch1** [appwrite/appwrite#5135] [Q&A] [answered]
Hi everyone, 
I'm switching my backend to AppWrite, I was previously using firebase and JWT for authentication and I had an endpoint called "refresh" to refresh the JWT token every hour (storing the current token in a DB so that you can only have 1 token active at a time and can't generate as many as you want). 

I have seen that AppWrite uses cookies and sessions (am I right?), should be a good practice to do the same thing here? Or I can keep the same cookie for example for 3 weeks and after that the user needs to log in again?

Thanks!

**Comment by stnguyen90** [comment, marked as answer]
You can use the same cookie and ask the user to log in after the cookie has expired. By default, the cookie lasts a year. You can change that my going to the Auth Settings for your project in the Appwrite Console. 

[This](https://github.com/appwrite/appwrite/discussions/3938#discussioncomment-3746725) might also be helpful for you.
